packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -881,8 +883,10 @@ class _AppBarState extends State<AppBar> {
           tooltip: MaterialLocalizations.of(context).openAppDrawerTooltip,
         );
       } else {
-        if (!hasEndDrawer && canPop)
+        final bool canHaveLeadingWithEndDrawer = !(hasEndDrawer && isFirstRoute && isEndDrawerOpen);
```

Comment:
if `isEndDrawerOpen` is true, the `hasEndDrawer` must also be true?

I think we can remove `hasEndDrawer`.

Reply:
Thanks, good point. I removed the `hasEndDrawer`.